packages/eslint-plugin/src/rules/no-invalid-css-map/__tests__/rule.test.ts

```diff
@@ -0,0 +1,639 @@
+          },
+        });
+      `,
+    },
+    {
+      name: 'function call to a default export',
+      errors: [
+        {
+          messageId: 'noFunctionCalls',
+        },
+      ],
+      options: [{ allowedFunctionCalls: [['@atlaskit/token', 'token']] }],
+      code: outdent`
+        import React from 'react';
+        import { cssMap } from '@compiled/react';
+        import token from '@atlaskit/token';
+
+        const styles = cssMap({
+          danger: {
+            color: token('red', 'blue'),
+            backgroundColor: 'red'
+          },
+        });
+      `,
+    },
+    {
+      name: 'function call to a local function',
+      errors: [
+        {
+          messageId: 'noFunctionCalls',
+        },
+      ],
+      options: [{ allowedFunctionCalls: [['@atlaskit/token', 'token']] }],
+      code: outdent`
+        import React from 'react';
+        import { cssMap } from '@compiled/react';
+
+        function token(...args) { return arguments.join(''); }
+
+        const styles = cssMap({
+          danger: {
+            color: token('red', 'blue'),
+            backgroundColor: 'red'
+          },
+        });
+      `,
+    },
+    {
+      name: 'function calls to forbidden functions with cheeky loophole (import from correct package)',
+      errors: [
+        {
+          messageId: 'noFunctionCalls',
+        },
+      ],
+      options: [{ allowedFunctionCalls: [['@atlaskit/token', 'token']] }],
+      code: outdent`
+        import React from 'react';
+        import { cssMap } from '@compiled/react';
+        import { invalidFunction as token } from '@atlaskit/token';
+
+        const styles = cssMap({
+          danger: {
+            color: token('red', 'blue'),
+            backgroundColor: 'red',
+          },
+          success: {
+            color: 'green',
+          },
+        });
+      `,
+    },
+    {
+      name: 'function calls to forbidden functions with cheeky loophole (import from wrong package)',
+      errors: [
+        {
+          messageId: 'noFunctionCalls',
+        },
+      ],
+      options: [{ allowedFunctionCalls: [['@atlaskit/token', 'token']] }],
+      code: outdent`
+        import React from 'react';
+        import { cssMap } from '@compiled/react';
+        import { invalidFunction as token } from 'obviously-the-wrong-package';
+
+        const styles = cssMap({
+          danger: {
+            color: token('red', 'blue'),
+            backgroundColor: 'red'
+          },
+          success: {
+            color: 'green',
+          },
+        });
+      `,
+    },
+    {
+      name: 'valid css map with variable in template strings',
+      errors: [
+        {
+          messageId: 'noNonStaticallyEvaluable',
+        },
+      ],
+      code: outdent`
+        import { firstFunction } from 'example';
+        import React from 'react';
+        import { cssMap } from '@compiled/react';
+        import { bap } from '@atlaskit/token';
```

Comment:
It's possible that `bap` is evaluable even if it's imported from another package/file.
It's programatically supported by Compiled, although we advocate colocating styles with their consumers,
You can find examples like the below in design system components.
```
import { N200, N300 } from '@atlaskit/theme/colors';

const colorMap = {
  light: {
    textColor: token('color.text.subtlest', N200),
    separatorColor: token('color.text.subtlest', N200),
  },
  dark: {
    textColor: token('color.text.subtlest', N300),
    separatorColor: token('color.text.subtlest', N300),
  },
};
```
, where N200 and N300 are from another package. We will likely continue to support that to ease the migration from emotion to Compiled.

Reply:
Ah that's a good point. Would it be ok to allow all imported variables but continue to forbid imported functions that aren't listed in `allowedFunctionCalls`?

Reply:
yup, that sounds good. 👍